modules/examiner/jsx/examiner.js

```diff
@@ -0,0 +1,252 @@
+import FilterForm from 'FilterForm';
+import formatColumn from './columnFormatter';
+import Loader from 'Loader';
+import Panel from 'Panel';
+/**
+ * Examiner Module Page.
+ *
+ * Serves as an entry-point to the module, rendering the whole react
+ * component page on load.
+ *
+ * Renders Examiner main page consisting of FilterTable and
+ * DataTable components.
+ *
+ * @author Victoria Foing, Zaliqa Rosli
+ * @version 1.0.0
+ *
+ * */
+class Examiner extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isLoaded: false,
+      Data: {},
+      formData: {
+        addName: null,
+        addRadiologist: null,
+        addSite: null
+      },
+      filter: {}
+    };
+
+    // Bind component instance to custom methods
+    this.fetchData = this.fetchData.bind(this);
+    this.pickElements = this.pickElements.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  /**
+   * Retrive data from the provided URL and save it in state
+   * Additionaly add hiddenHeaders to global loris vairable
+   * for easy access by columnFormatter.
+   */
+  fetchData() {
+    const self = this;
+    $.ajax(this.props.DataURL, {
+      method: "GET",
+      dataType: 'json',
+      success: function(data) {
+        loris.hiddenHeaders = data.hiddenHeaders ? data.hiddenHeaders : [];
+        self.setState({
+          Data: data,
+          isLoaded: true
+        });
+      },
+      error: error => {
+        console.error(error);
+      }
+    });
+  }
+
+  pickElements(json, keys) {
+    let subset = {};
+    keys.forEach(function(key) {
+      if (json.hasOwnProperty(key)) {
+        subset[key] = json[key];
+      }
+    });
+    return subset;
+  }
+
+  setFormData(formElement, value) {
+    let formData = this.state.formData;
+    formData[formElement] = value;
+    this.setState({
+      formData: formData
+    });
+  }
+
+  updateFilter(filter) {
+    this.setState({filter});
+  }
+
+  resetFilters() {
+    this.refs.examinerFilter.clearFilter();
```

Comment:
@davidblader what do i replace refs with here?

Reply:
https://reactjs.org/docs/refs-and-the-dom.html#callback-refs